### PR TITLE
Centralize all validation exclusions

### PR DESCRIPTION
**Pull request by geoknee**
Should make them easier to audit. 

TODO: 
- [x] centralize remaining exclusions, may need to break some tests up

### CODE REVIEW

File: validation/exclusions_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		11155421: true, // sepolia-dev-0/oplabs-devnet-0   (no public endpoint)
+		11763072: true, // sepolia-dev-0/base-devnet-0     (no public endpoint)
+	},
+	"L2OO_Params": {
+		10:        true, // OP mainnet - Upgraded to fault proofs
+		999999999: true, // sepolia/zora  Incorrect finalizationPeriodSeconds, 604800 is not within bounds [12 12]
+		1740:      true, // sepolia/metal Incorrect finalizationPeriodSeconds, 604800 is not within bounds [12 12]
+		919:       true, // sepolia/mode  Incorrect finalizationPeriodSeconds, 180 is not within bounds [12 12]
+		11155420:  true, // sepolia/op No L2OO because this chain uses Fault Proofs https://github.com/ethereum-optimism/superchain-registry/issues/219
+		11155421:  true, // oplabs-sepolia-devnet-0 No L2OO because this chain uses Fault Proofs https://github.com/ethereum-optimism/superchain-registry/issues/219
```
Comment (geoknee) [inline validation/exclusions_test.go:35]: Now these exclusions are in one place, it is easier to see how we can have fewer of them. If the standard config requires at least fault proofs, we ought to be able to drop this test entirely (and a bunch of other code besides).

File: validation/exclusions_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		84532:     true, // base-sepolia (incorrect challenger)
+		7777777:   true, // zora (incorrect challenger)
+		1750:      true, // metal (incorrect challenger)
+		919:       true, // mode sepolia (incorrect challenger)
+		999999999: true, // zora sepolia (incorrect challenger)
+		34443:     true, // mode (incorrect challenger)
+	},
+	"L2_Security_Config": {
+		11155421: true, // sepolia-dev-0/oplabs-devnet-0   No Public RPC declared
+		11763072: true, // sepolia-dev-0/base-devnet-0     No Public RPC declared
```
Comment (geoknee) [inline validation/exclusions_test.go:50]: We probably want a neater way of handling devnets, as these come up often in this exclusion list.

File: validation/validation_test.go [hunk 3]
```diff
@@ -63,15 +63,15 @@ func testUniversal(t *testing.T, chain *ChainConfig) {
...
-		testDataAvailability(t, chain)
-		testResourceConfig(t, chain)
-		testL2OOParams(t, chain)
-		testGasLimit(t, chain)
-		testGasPriceOracleParams(t, chain)
+		t.Run("Data Availability", func(t *testing.T) { testDataAvailability(t, chain) })
+		t.Run("Resource Config", func(t *testing.T) { testResourceConfig(t, chain) })
+		t.Run("L2OO Params", func(t *testing.T) { testL2OOParams(t, chain) })
+		t.Run("Gas Limit", func(t *testing.T) { testGasLimit(t, chain) })
+		t.Run("GPO Params", func(t *testing.T) { testGasPriceOracleParams(t, chain) })
```
Comment (bitwiseguy) [inline validation/validation_test.go:70]: What is the advantage of wrapping these test functions with the `t.Run()` method?

**Review by bitwiseguy — APPROVED** [review]
Looks good. Just left one question

File: validation/validation_test.go (hunk 3 above)
Comment (geoknee) [reply validation/validation_test.go:70]: This is the only way we can exclude a chain from some of these tests but not others. With the `.Run()` syntax, the test is executed as a _named_ sub test, which we can target in the `exclusions_test.go` file.

**Pull request merged**